addons/mail/models/discuss/res_partner.py

```diff
@@ -98,7 +98,7 @@ def get_mention_suggestions_from_channel(self, channel_id, search, limit=8):
         )
         store = Store(members, fields={"channel": [], "persona": []})
         for p in partners:
-            store.add(p,{
-                "groups_id": [("ADD", next((group.id for group in p.user_ids.groups_id if group.id == channel.group_public_id.id), None))]
```

Comment:
`(p.user_ids.groups_id & channel.group_public_id).ids` instead of next(...)

Reply:
And wrap the loop in if `channel.group_public_id` to avoid doing this processing for channels that don't need it.